src/main/java/com/minecolonies/coremod/inventory/ContainerItemHandler.java

```diff
@@ -0,0 +1,98 @@
+package com.minecolonies.coremod.inventory;
+
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.inventory.Container;
+import net.minecraft.inventory.IInventory;
+import net.minecraft.inventory.Slot;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.items.IItemHandler;
+import net.minecraftforge.items.SlotItemHandler;
+
+import javax.annotation.Nullable;
+
+/**
+ * Created by asie on 2/16/17.
+ */
+public class ContainerItemHandler extends Container {
+    private final IItemHandler handler;
+    private final int numRows;
+
+    public ContainerItemHandler(IInventory playerInventory, IItemHandler chestInventory, EntityPlayer player)
+    {
+        this.handler = chestInventory;
+        this.numRows = chestInventory.getSlots() / 9;
+        int i = (this.numRows - 4) * 18;
+
+        for (int j = 0; j < this.numRows; ++j)
+        {
+            for (int k = 0; k < 9; ++k)
+            {
+                this.addSlotToContainer(new SlotItemHandler(chestInventory, k + j * 9, 8 + k * 18, 18 + j * 18));
+            }
+        }
+
+        for (int l = 0; l < 3; ++l)
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Assign this magic number 3 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)